test/unit/yesno.spec.ts

```diff
@@ -220,6 +220,36 @@ describe('Yesno', () => {
       expect(yesno.intercepted()).to.have.lengthOf(1);
     });
 
+    it('should support out of order mocks', async () => {
+      yesno.mock([createMock(), createMock(), createMock()]);
+      yesno
+        .matching({
+          request: {
+            path: '/test',
+          },
+        })
+        .respond({
+          body: 'fizbaz',
+          statusCode: 200,
+        });
+
+      expect(yesno.intercepted()).to.have.lengthOf(0);
+
+      // consumes first mock
+      let response = await requestTestServer();
+      expect(response).to.eql('foobar');
+
+      // consumes matched mock
+      response = await requestTestServer({ uri: 'http://localhost/test' });
+      expect(response).to.eql('fizbaz');
+      expect(yesno.intercepted()).to.have.lengthOf(2);
+
+      // consumes third mock
+      response = await requestTestServer();
+      expect(response).to.eql('foobar');
```

Comment:
I'd like to know which of the three original mocks is invoked here. Is it the second or third one? Maybe you can do this
```
[createMock({ response: { body: "a" }), createMock({ response: { body: "b" }, createMock({ response: { body: "c" }))]
```
Then assert each time you invoke `requestTestServer` like you did here `expect(response).to.eql('fizbaz');`